exmg-sortable.ts

```diff
@@ -0,0 +1,395 @@
+    ) {
+      this.insertAtTarget(this.draggedElement!, target);
+    }
+  }
+
+  /**
+   * Fast and simple hit test to check whether the center of a node intersects with the rectangle of any of the
+   * given targets. Returns an array of matches.
+   *
+   * @param {HTMLElement} node
+   * @param {Array} targets
+   * @return {Array<HTMLElement>} matches
+   */
+  private hitTest(node:HTMLElement, targets: HTMLElement[]): HTMLElement[] {
+    const {left: l, top: t, width: w, height: h} = node.getBoundingClientRect();
+    const x = l + (w / 2);
+    const y = t + (h / 2);
+
+    return targets.filter(item => {
+      const {left, right, top, bottom} = item.getBoundingClientRect();
+      return ! (x < left || x > right || y < top || y > bottom);
+    });
+  }
+
+  /**
+   * Returns a boolean indicating whether the node is currently in an animation.
+   *
+   * @param {HTMLElement} node
+   * @returns {boolean} isAnimating
+   */
+  private isAnimating(node: HTMLElement): boolean {
+    return this.animatedElements.indexOf(node) > -1;
+  }
+
+  private reset(): void {
+    if (this.draggedElementClone !== undefined && this.draggedElementClone.parentNode !== null) {
+      this.draggedElementClone.parentNode.removeChild(this.draggedElementClone);
+    }
+
+    if (this.draggedElement && this.draggedElement.parentNode && this.draggedElementOrigin) {
+      this.draggedElement.classList.remove(this.draggedClass);
+      this.draggedElement.parentNode.insertBefore(this.draggedElement, this.draggedElementOrigin);
+    }
+
+    delete this.draggedElementClone;
+    delete this.draggedElement;
+
+    this.sortableNodes = [];
+    this.animatedElements = [];
+    this.dragRequestPending = false;
+    this.style.webkitUserSelect = 'text';
+  }
+
+  /**
+   * Triggers a CSS animation on a node with the given dx and dy. Used following dom updates to make it appear as
+   * if nodes animate from their old to their new position in the dom.
+   *
+   * @param {Node} node
+   * @param {number} dx
+   * @param {number} dy
+   */
+  private animateNode(node: HTMLElement, dx = 0, dy = 0): void {
+    if (!node.animate) {
+      return;
+    }
+
+    // keep a stack of currently animating nodes to exclude as drag & drop targets.
+    this.animatedElements.push(node);
+
+    Object.assign(node.style, {
+      willChange: 'transform',
+    });
+
+    // animate from dx/dy (old node position) to none (new node position)
+    this.animationPromise = new Promise((resolve) => {
+      node.animate([
+        {transform: `translate3d(${dx}px, ${dy}px, 0)`},
+        {transform: 'none'},
+      ], this.animationTiming).addEventListener('finish', () => {
+        const index = this.animatedElements.indexOf(node);
+        Object.assign(node.style, {
+          willChange: 'initial',
+        });
+        if (index !== -1) {
+          // splice out when done to unlock as a valid target
+          this.animatedElements.splice(index, 1);
+        }
+
+        resolve();
```

Comment:
resolved value is void - it means that this.animationPromise should be type Promise<void> instead Promise<any>

Reply:
done